pkg/apis/extensions/v1beta1/types.go

```diff
@@ -1036,3 +1037,58 @@ type PodSecurityPolicyList struct {
 	// Items is a list of schema objects.
 	Items []PodSecurityPolicy `json:"items"`
 }
+
+// Template contains the inputs needed to produce a Config.
+type Template struct {
```

Comment:
"ParameterizedList" was suggested as potential name during the template discussions.  I like it from the perspective that it accurately explains what this object is.  I dislike it from the perspective that it's not user-friendly for the context when you have end-users creating these objects.